server/routes/referrals/referralsController.test.ts

```diff
@@ -1112,45 +1115,95 @@ describe('POST /referrals/:referralId/service-category/:service-category-id/desi
+    beforeEach(() => {
+      const serviceCategory = serviceCategoryFactory.build({
+        id: 'b33c19d1-7414-4014-b543-e543e59c5b39',
+        desiredOutcomes,
+        name: 'social inclusion',
+      })
+      const referral = draftReferralFactory.serviceCategoriesSelected([serviceCategory.id]).build()
 
-    interventionsService.getDraftReferral.mockResolvedValue(referral)
-    interventionsService.getServiceCategory.mockResolvedValue(serviceCategory)
+      interventionsService.getDraftReferral.mockResolvedValue(referral)
+      interventionsService.getServiceCategory.mockResolvedValue(serviceCategory)
+    })
+
+    it('updates the referral on the backend and redirects to the complexity level page', async () => {
+      await request(app)
+        .post('/referrals/1/service-category/b33c19d1-7414-4014-b543-e543e59c5b39/desired-outcomes')
+        .type('form')
+        .send({ 'desired-outcomes-ids[]': [desiredOutcomes[0].id, desiredOutcomes[1].id] })
+        .expect(302)
+        .expect('Location', '/referrals/1/complexity-level')
+
+      expect(interventionsService.setDesiredOutcomesForServiceCategory).toHaveBeenCalledWith('token', '1', {
+        serviceCategoryId: 'b33c19d1-7414-4014-b543-e543e59c5b39',
+        desiredOutcomesIds: [desiredOutcomes[0].id, desiredOutcomes[1].id],
+      })
+    })
   })
 
-  it('updates the referral on the backend and redirects to back to the form page', async () => {
-    await request(app)
-      .post('/referrals/1/service-category/b33c19d1-7414-4014-b543-e543e59c5b39/desired-outcomes')
-      .type('form')
-      .send({ 'desired-outcomes-ids[]': [desiredOutcomes[0].id, desiredOutcomes[1].id] })
-      .expect(302)
-      .expect('Location', '/referrals/1/form')
+  describe('for a cohort referral', () => {
+    beforeEach(() => {
+      const serviceCategory = serviceCategoryFactory.build({
+        id: 'b33c19d1-7414-4014-b543-e543e59c5b39',
+        desiredOutcomes,
+        name: 'social inclusion',
+      })
+      const referral = draftReferralFactory
+        .serviceCategoriesSelected([serviceCategory.id, serviceCategoryFactory.build().id])
+        .build()
 
-    expect(interventionsService.setDesiredOutcomesForServiceCategory).toHaveBeenCalledWith('token', '1', {
-      serviceCategoryId: 'b33c19d1-7414-4014-b543-e543e59c5b39',
-      desiredOutcomesIds: [desiredOutcomes[0].id, desiredOutcomes[1].id],
+      interventionsService.getDraftReferral.mockResolvedValue(referral)
+      interventionsService.getServiceCategory.mockResolvedValue(serviceCategory)
+    })
+
+    it('updates the referral on the backend and redirects to back to the form page', async () => {
+      await request(app)
+        .post('/referrals/1/service-category/b33c19d1-7414-4014-b543-e543e59c5b39/desired-outcomes')
+        .type('form')
+        .send({ 'desired-outcomes-ids[]': [desiredOutcomes[0].id, desiredOutcomes[1].id] })
+        .expect(302)
+        .expect('Location', '/referrals/1/form')
```

Comment:
how come it doesn't redirect to the desired outcomes page for the next service category?

Reply:
The designs haven't quite been finalised yet. I think it'll go to complexity levels next (then to the next service category), but I wanted to wait until we've refactored the complexity levels stuff in the next ticket before hooking up the cohort services properly to avoid further conditionals here.